web/src/test/java/com/bazaarvoice/emodb/web/scanner/ScanUploaderTest.java

```diff
@@ -307,11 +309,15 @@ public S3ScanWriter answer(InvocationOnMock invocation)
         TableSet tableSet = mock(TableSet.class);
         StashStateListener stashStateListener = mock(StashStateListener.class);
         ScanCountListener scanCountListener = mock(ScanCountListener.class);
+        DataCenters dataCenters = mock(DataCenters.class);
+        DataCenter dataCenter1 = mockDataCenter("us-east", "http://emodb.cert.us-east-1.nexus.bazaarvoice.com:8081", "http://emodb.cert.us-east-1.nexus.bazaarvoice.com:8080");
+        when(dataCenters.getSelf()).thenReturn(dataCenter1);
+        when(dataCenters.getAll()).thenReturn(ImmutableList.of(dataCenter1));
 
         ScanWorkflow scanWorkflow = new InMemoryScanWorkflow();
         ScanStatusDAO scanStatusDAO = new InMemoryScanStatusDAO();
         // Create the instance and run the upload
-        ScanUploader scanUploader = new ScanUploader(dataTools, scanWorkflow, scanStatusDAO, stashStateListener);
+        ScanUploader scanUploader = new ScanUploader(dataTools, scanWorkflow, scanStatusDAO, stashStateListener, dataCenters, metricRegistry);
```

Comment:
This unit test won't work because `ScanUploader` needs to have `DefaultSystemManager` injected.  Otherwise the default implementation always tries to open a connection to a remote host which does not exist.  With injection you can provide a mock and verify the correct calls are made.